docker/exporter/export_runner.py

```diff
@@ -74,6 +78,39 @@ def spawn_ecosystem_exporter(work_dir: str, bucket: str, eco: str):
     logging.error('Export of %s failed with Exit Code: %d', eco, return_code)
 
 
+def aggregate_all_vulnerabilities(work_dir: str, export_bucket: str):
+  """
+  Aggregates vulnerability records from each ecosystem into a single zip
+  file and uploads it to the export bucket.
+  """
+  zip_file_name = 'all.zip'
+  output_zip = os.path.join(work_dir, zip_file_name)
+  all_vulns = {}
+
+  for ecosystem_dir, _, files in os.walk(work_dir):
+    if ecosystem_dir == work_dir:
+      continue
+    for vuln_filename in files:
+      if not vuln_filename.endswith('.json'):
+        continue
+      all_vulns[vuln_filename] = os.path.join(ecosystem_dir, vuln_filename)
+
+  with z.ZipFile(output_zip, 'a') as all_zip:
+    for vuln_filename in sorted(all_vulns):
+      file_path = all_vulns[vuln_filename]
+      all_zip.write(file_path, os.path.basename(file_path))
+
+  storage_client = storage.Client()
+  bucket = storage_client.get_bucket(export_bucket)
+  upload_single(bucket, output_zip, zip_file_name)
+
+
+def clean_work_dir(work_dir: str):
```

Comment:
Rather than using clean_work_dir, let's just create a temporary directory similar to how it was done in the individual exporter runners, since this is what it's used for. (just make sure to set the environment variable for the temporary directory to the working directory so it always writes to the ssd)

And pass the temporary directory path into the individual exporter runs as the work dir.

Reply:
updated